### PR TITLE
Move create collection to RP

### PR DESCRIPTION
**Pull request by victor-meng**
- refactored logic in AddCollectionPane to call `createCollection` for all API types
- manually added `analyticalStorageTtl` in `SqlContainerResource` type
- fixed wrong parameter types in `createDatabase`

### CODE REVIEW

File: src/Common/dataAccess/createCollection.ts [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+    userContext.resourceGroup,
+    userContext.databaseAccount.name,
+    params.databaseId,
+    params.collectionId,
+    rpPayload
+  );
+  return createResponse && (createResponse.properties.resource as DataModels.Collection);
+};
+
+const createGremlineGraph = async (params: DataModels.CreateCollectionParams): Promise<DataModels.Collection> => {
```
Comment (languy) [inline src/Common/dataAccess/createCollection.ts:232]: nit: typo in Gremlin. But I would rename: `createGraph`.

File: src/Common/dataAccess/createCollection.ts [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+    userContext.subscriptionId,
+    userContext.resourceGroup,
+    userContext.databaseAccount.name,
+    params.collectionId,
+    rpPayload
+  );
+  return createResponse && (createResponse.properties.resource as DataModels.Collection);
+};
+
+const constructRpOptions = (params: DataModels.CreateDatabaseParams): ARMTypes.CreateUpdateOptions => {
```
Comment (languy) [inline src/Common/dataAccess/createCollection.ts:322]: Is this logic covered in the unit tests?

File: src/Common/dataAccess/createCollection.ts [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+    id: params.collectionId,
+    partitionKey: params.partitionKey || undefined,
+    indexingPolicy: params.indexingPolicy || undefined,
+    uniqueKeyPolicy: params.uniqueKeyPolicy || undefined,
+    analyticalStorageTtl: params.analyticalStorageTtl
+  } as ContainerRequest; // TODO: remove cast when https://github.com/Azure/azure-cosmos-js/issues/423 is fixed
+  const collectionOptions: RequestOptions = {};
+  const createDatabaseBody: DatabaseRequest = { id: params.databaseId };
+
+  if (params.databaseLevelThroughput) {
```
Comment (languy) [inline src/Common/dataAccess/createCollection.ts:351]: Is this logic covered in the unit tests?

**Review by languy — APPROVED** [review]

File: src/Common/dataAccess/createCollection.ts (hunk 1 above)
Comment (victor-meng) [reply src/Common/dataAccess/createCollection.ts]: Done

File: src/Common/dataAccess/createCollection.ts (hunk 3 above)
Comment (victor-meng) [reply src/Common/dataAccess/createCollection.ts:351]: Added unit tests for `constructRpOptions` function

File: src/Common/dataAccess/createCollection.ts (hunk 2 above)
Comment (victor-meng) [reply src/Common/dataAccess/createCollection.ts]: Added unit tests for `constructRpOptions` function

**Pull request merged**